### PR TITLE
Add Support for Directories in RV Packages SupportPath

### PR DESCRIPTION
**Pull request by shannonyearwood**
### Summarize your change.
When creating a package you can list folders with files that should be installed in the SupportPath Plugins folder rather than listing each file individually

### Describe the reason for the change.
It should be easier/quicker to create packages

### Describe what you have tested and on which operating system.
MacOS

### Add a list of changes, and note any that might need special attention during the review.
- Packages are zipped with directory structure
- All files in package and its directories are zipped
- The app will traverse the PACKAGE file and see the list of folders specified. Based on this, all files under the folder and its subdirectories will be added to location provided.

N.B. When creating PACKAGE file if you want to use new behaviour it takes the following format:
folders:
- folder: [FolderName]
   location: [LocationName]

### If possible, provide screenshots.

### CODE REVIEW

File: cmake/macros/rv_stage.cmake [hunk 1]
```diff
@@ -456,7 +456,7 @@ FUNCTION(rv_stage)
       COMMENT "Creating ${_package_filename} ..."
       OUTPUT ${_package_filename}
       DEPENDS ${_files} ${_package_file}
-      COMMAND zip -v -j ${_package_filename} ${_files} ${_package_file}
+      COMMAND zip -v -J ${_package_filename} ${_files} ${_package_file}
```
Comment (geffrak) [inline cmake/macros/rv_stage.cmake:459]: I think you can remove the `-J` as it was a typo for `-j` and you implemented the required to not required `-j` anymore.

File: src/lib/app/RvPackage/PackageManager.cpp [hunk 2]
```diff
@@ -1070,6 +1110,34 @@ PackageManager::loadPackageInfo(const QString& infileNonCanonical)
 				yaml_event_delete(&input_event);
 				yaml_parser_delete(&parser);
 
+        // Standardize the file and directory paths
+        for (auto& fileName : package.files) {
+          fileName.replace("\\", "/");
```
Comment (geffrak) [inline src/lib/app/RvPackage/PackageManager.cpp:1115]: There's no native Qt function for this operation?

**Review by geffrak — APPROVED** [review]
Looks great. Does it work for both install and uninstall?

**Comment by geffrak** [comment]
What happens if your packages contain a folder but don't have any `folder` directive to handle it?

File: src/lib/app/RvPackage/PackageManager.cpp [hunk 3]
```diff
@@ -910,7 +923,14 @@ PackageManager::loadPackageInfo(const QString& infileNonCanonical)
                 temp[255] = 0;
                 files.push_back(temp);
                 QString f = temp;
-                if (f != "PACKAGE") m_packages.back().files.push_back(f);
+                // Check if it's a directory by checking if it ends with a '/'
+                if (f.endsWith("/"))
```
[the remark this answers is not in the document]
Comment (geffrak) [reply src/lib/app/RvPackage/PackageManager.cpp:927]: There might indeed be a Qt native call to check if it's a directory

**Review by rogernelson — APPROVED** [review]

**Pull request merged**